commerce_hubspot.module

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/**
+ * @file
+ * Hooks and functionality for the Commerce HubSpot module.
+ */
+
+use Drupal\commerce_hubspot\Event\SyncEntityTypesEvent;
+use Drupal\Core\Routing\RouteMatchInterface;
+use Drupal\Core\Entity\EntityInterface;
+
+const COMMERCE_HUBSPOT_LOGGER_CHANNEL = 'commerce_hubspot';
+
+/**
+ * Implements hook_help().
+ */
+function commerce_hubspot_help($route_name, RouteMatchInterface $route_match) {
+  switch ($route_name) {
+    case 'help.page.commerce_hubspot':
+      $output = '';
+      $output .= '<h3>' . t('About') . '</h3>';
+      $output .= '<p>' . t(
+          'The Commerce Hubspot module integrates Drupal
+        Commerce with Hubspot. It syncs data with the <a target="_blank" href=":hubspot_api">Hubspot API</a>
+        any time a user, order, or product is created or updated.', [
+          ':hubspot_api' => 'https://www.hubspot.com',
+        ]
+      ) . '</p>';
+
+      return $output;
+  }
+}
+
+/**
+ * Implements hook_entity_update().
+ */
+function commerce_hubspot_entity_update(EntityInterface $entity) {
+  $entities_to_sync = [
+    'order',
+    'order_item',
+    'product_variation',
+    'profile',
+    'user'
+  ];
+
+  // Dispatch our sync entity types event.
+  $event = new SyncEntityTypesEvent($entity, $entities_to_sync);
+  $event_dispatcher = \Drupal::service('event_dispatcher');
+  $event_dispatcher->dispatch(SyncEntityTypesEvent::EVENT_NAME, $event);
+
+  if (!in_array($entity->getEntityType()->id(), $entities_to_sync)) {
+    return;
+  }
+
+  $queue_factory = \Drupal::service('queue');
+  $queue = $queue_factory->get('commerce_hubspot_entity_sync_to');
+  $queue->createItem($entity);
+}
```

Comment:
**CHANGE REQUEST**

We only need to the store the entity ID as the queue item's data. We can then load the full entity during the queue item processing; it's happen on the background, performance is less important there.

I'm not sure what this will do if you pass the full entity to `createItem`, is it going to store the full entity serialized as the queue item's data? Apart from performance issues (storing just the ID will be a much faster query) there is the chance of having stale data when processing the item e.g. we queue the entity with certain field values, then somebody changes the entity, then the queue starts processing it but it loads the queued entity and syncs the old values. That all is avoided if we just store the entity ID and the entity is loaded directly by the queue worker.